webapp/src/ts/modules/contacts/contacts-content.component.ts

```diff
@@ -153,4 +199,182 @@ export class ContactsContentComponent implements OnInit, OnDestroy {
     this.tasksTimeWindowWeeks = weeks;
     this.taskEndDate = weeks ? moment().add(weeks, 'weeks').format('YYYY-MM-DD') : null;
   }
+
+  private async setRightActionBar() {
+    await this.setChildTypesBySelectedContact();
+    await this.setUserSettings();
+    await this.setSettings();
+
+    this.globalActions.setRightActionBar({
+      relevantForms: [], // This disables the "New Action" button until forms load
+      sendTo: this.selectedContact?.type?.person ? this.selectedContact?.doc : '',
+      canDelete: this.canDeleteContact,
+      canEdit: this.sessionService.isAdmin() || this.userSettings?.facility_id !== this.selectedContact?.doc?._id,
+      openContactMutedModal: (form) => this.openContactMutedModal(form),
+      openSendMessageModal: (sendTo) => this.openSendMessageModal(sendTo)
+    });
+
+    this.subscribeToAllContactXmlForms();
+    this.subscribeToSelectedContactXmlForms();
+  }
+
+  private setUserSettings() {
+    if (this.userSettings) {
+      return;
+    }
+    return this.userSettingsService
+      .get()
+      .then(userSettings => this.userSettings = userSettings)
+      .catch(e => console.error('Error fetching user settings', e));
+  }
+
+  private setSettings() {
+    if (this.settings) {
+      return;
+    }
+    return this.settingsService
+      .get()
+      .then(settings => this.settings = settings)
+      .catch(e => console.error('Error fetching settings', e));
+  }
+
+  private async setChildTypesBySelectedContact() {
```

Comment:
Same as above. Please change this to use then chains.

Reply:
Actually prefer this one. it's working fine, readable, keeping code flat. We're going to start moving to async/await so files will have mix. :)

Reply:
Some things don't support async/await and probably never will. So I'm reluctant. 

For example, you can't `await PouchDB().changes()`, because it's not a promise. It's an event emitter with a `then` property. 
This is just an example.

Reply:
Okay, that can stay as promise :)